### PR TITLE
Optional metadata fields when managing an application

### PR DESCRIPTION
**Pull request by AramAlsabti**
This PR adds optional metadata fields on the create and update application page. For now, none of the fields affect any of the pre-existing logic.

A new device type, "Other", has been added. To not confuse it with a real device type, a distinction is made between application device types (the new metadata field) and IoT device types (HTTP, LoRaWAN, Sigfox).

[Related PR](https://github.com/OS2iot/OS2IoT-backend/pull/152)

### CODE REVIEW

File: src/app/shared/components/forms/form-body-application/form-body-application.component.html
Comment (GufCab) [inline src/app/shared/components/forms/form-body-application/form-body-application.component.html:37]: Date is not validated - I can put in non-date string in the text field and it will just not show up in the DB.
Should show error if date cannot be validated

File: src/app/shared/components/metadata-details/metadata-details.component.html
Comment (GufCab) [inline src/app/shared/components/metadata-details/metadata-details.component.html:1]: Should be hidden if status is set to None. As it is now, when nothing is chosen for status, this is shown:
image

**Review by GufCab — COMMENTED** [review]
Minor comments, can be merged after fixing

**Comment by AramAlsabti** [comment]
Fixes have been applied. This cannot be merged until the related backend PR is approved.

**Review by GufCab — APPROVED** [review]
I have also approved the backend PR. Merge when ready.

**Pull request merged**
